service/controller/app/v1/resource/chartoperator/create.go

```diff
@@ -58,7 +58,7 @@ func (r Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not find chart-perator release %#q in tenant cluster", release))
 			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("installing chart-operator release %#q in tenant cluster", release))
 
-			err = r.installChartOperator(ctx, cr, cc.HelmClient)
```

Comment:
Instead of giving it as parameters, I would propagate context inside the function.